api/v1beta1/flowcollector_types.go

```diff
@@ -233,6 +233,24 @@ type FlowCollectorKafka struct {
 	TLS ClientTLS `json:"tls"`
 }
 
+type FlowCollectorIPFIXReceiver struct {
+	//+kubebuilder:default:=""
+	// address of the ipfix external receiver
+	TargetHost string `json:"address"`
+
+	//+kubebuilder:default:=""
+	// port for the ipfix external receiver
+	TargetPort int `json:"port"`
+
+	// Transport protocol (tcp/udp) to be used for the IPFIX connection, defaults to tcp
+	//+optional
+	Transport string `json:"transport,omitempty"`
```

Comment:
we should use an `enum` here as values are either `tcp` or `udp`.

Check `FlowCollectorExporter` `Type` for example

Reply:
Done!

Reply:
I think you can remove `unionDiscrimitator`  here: it is useful if there were specific sub-settings to TCP or UDP, but that's not the case here?